finagle-netty4-http/src/main/scala/com/twitter/finagle/netty4/http/package.scala

```diff
@@ -142,11 +143,14 @@ package object http {
         if (autoContinue)
           pipeline.addLast("expectContinue", new HttpServerExpectContinueHandler)
 
-        // no need to handle expect headers in the finexLenAggregator since we have the task
+        // buffer size for messages with fixed length
+        val fixedLengthBuffer = aggregateIfLessThan.min(maxRequestSize)
+
+        // no need to handle expect headers in the fixedLenAggregator since we have the task
         // specific HttpServerExpectContinueHandler above.
         pipeline.addLast(
           "fixedLenAggregator",
-          new FixedLengthMessageAggregator(maxRequestSize, handleExpectContinue = false)
```

Comment:
@bryce-anderson and I are talked about this. It seems like we need to keep `maxRequestSize` in place as it's also get passed to a supper class, `FinagleHttpObjectAggregator`, which in turn, takes care about handling oversized messages (think HTTP 413). Changing this param to a different value, for example 128kb, means your HTTP server would be returning 413 on each payload bigger than 128kb.

We really need two different parameters here. One for the super-class (object aggregator), one for aggregate-if-less-then.

Reply:
I may be missing something, but I think, we do not need new parameter.

If message is bigger than `aggregateIfLessThen`, then there's nothing to aggregate. Messages fully bypass this handler.

Your concern would be reasonable if we used `FinagleHttpObjectAggregator`, but this class adds additional checks on top of it.

To sum up, this class is only doing something in following cases
* message length is known and it is less than `maxContentLength`
* message length is unknown, but some heuristics allow us to conclude that message size is `0` (and we aggregate only to have request with `isChunked == false`, so no allocations on buf are needed).

There are already some unit tests for the `FixedLengthMessageAggregator` that demonstrate how it works if `maxContentLength` is 0. Also all end2end tests now use `FixedLengthMessageAggregator` with `maxContentLength == 0`  (and all pass). I can add more tests for other cases, if you have any in mind.

Reply:
Oh the `shouldAggregate` check. Yeah, I think you're right.

Maybe it's worth mentioning in the scaladoc for `withStreaming` method that `aggregateIfLessThan` could be replaced with `maxRequestSize` if the later is smaller.

Also, what do you think about omitting the handler all-together if `maxContentLength` is zero?

Reply:
> Maybe it's worth mentioning in the scaladoc for withStreaming method that aggregateIfLessThan could be replaced with maxRequestSize if the later is smaller.

Sorry, I'm not sure I follow.
To my understanding, `maxRequestSize` limits request size and in case if streaming is enabled it is enforced by `PayloadSizeHandler`, not by `FixedMessageLengthAggregator`.
If `maxRequestSize < aggregateIfLessThan`, then the message should be rejected even before it reaches `FixedMessageLengthAggregator`. In any case, the behavior remains consistent, and if message length `> maxRequestSize`, then 413 must be returned regardles of the `aggregateIfLessThan`.

> Also, what do you think about omitting the handler all-together if maxContentLength is zero?

We can't omit the handler completely if `maxContentLength == 0` because it handles also 2 cases that are not impacted by `maxContentLength`:
* responses with specific codes that are guaranteed to have 0 length
* requests without `Content-Length` and without `Transfer-Encoding: chunked`

If we omit this handler, those 2 cases will end up with chunked `Request` and `Response` objects.
This is not critical perhaps, but this is a change in behavior.

Another option might be to split the handler in two handlers: one to handle messages that have body and the other to handle messages that do not have body. So we could omit the first one if `aggregateIfLessThan == 0`. But frankly, at this stage, I do not see much benefit in this. Once you start thinking about `FixedLengthMessage` and how we define it (in scaladoc), this handler makes perfect sense as is (at least to me).

Reply:
> We can't omit the handler completely if maxContentLength == 0 because it handles also 2 cases that are not impacted by maxContentLength:

On the other hand, if developer enables streaming, `isChunked == true` should be expected. IDK, honestly. This is tough design decision, and I am not sure I have enough exprience with finagle to be leaning towards any of the options.

Reply:
Yeah, I'm with you on the understanding front. What I mean is, when users configure `withStreaming(true, 10.mb)` they expect fixed-length messages up to 10mb to be aggregated. Yet, as we take min between max-request-size and aggregate-if-less-than`, we actually cap it at 5mb (max-request-size default). Let's just document that we take the minimum between the two.

Also, really good point about messages w/o bodies that still needs to be "aggregated".

Reply:
Ok, I got it now! Will do the fix a bit later today. Hopefully together with some new e2e tests.

Reply:
I think I prefer if we prioritize `Http.withStreaming` parameters over the `maxContentLength`, i.e. `maxRequestSize` is ignored when `aggregateIfLessThan > 0`. It changes behavior, but I'm concerned about complicating configuration for the user. To me, this seems to be a good default.

Actually, I wonder if we can make a bolder change and ignore `maxRequestSize` when `Http.streaming` is enabled.

Reply:
I don't think we should ignore it. There is `PayloadSizeFilter` that would reject payloads bigger than a max-size so it doesn't really make sense to aggregate more than that anyway.